sql-parser/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLStatementSQLVisitor.java

```diff
@@ -350,7 +350,7 @@ private Optional<String> findBinaryOperator(final AExprContext ctx) {
     }
     
     private BinaryOperationExpression createPatternMatchingOperationSegment(final AExprContext ctx) {
-        String operator = getOriginalText(ctx.patternMatchingOperator());
+        String operator = getOriginalText(ctx.patternMatchingOperator()).toUpperCase();
```

Comment:
Can you modify the kernel logic instead of sql parse logic?

Reply:
Do you mean to modify the code logic of the `EncryptPredicateRightValueTokenGenerator@getEncryptedValues` function to fix this bug? I don't think it will work because many code logics in the encrypt rewrite assume that these tokens are uppercase letters by default. If you make such modifications, it will deviate significantly from the previous code logic.

https://github.com/apache/shardingsphere/blob/5.3.1/features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptPredicateRightValueTokenGenerator.java#L98
```diff
- if (encryptCondition instanceof EncryptBinaryCondition && "LIKE".equals(((EncryptBinaryCondition) encryptCondition).getOperator())) {
+ if (encryptCondition instanceof EncryptBinaryCondition && "LIKE".equalsIgnoreCase(((EncryptBinaryCondition) encryptCondition).getOperator())) {
```

Reply:
Yes, the SQL parsing engine is a separate module that should not change the original content of the SQL. The kernel is where the SQL logic is handled, and we should do it in the right place.

Reply:
Shouldn't the BinaryOperationExpression operator be the operation pair of a binary operation expression?
In SQL, logical operators `AND OR NOT LIKE` can be converted to uppercase, while other operators (comparison operators, arithmetic operators) have no effect after executing toUpperCase().
Furthermore, I noticed that the operator in BinaryOperationExpression is directly assigned in MySQLStatementSQLVisitor, and I made this change based on its code.

I have just started reading the Shardingsphere code recently, and if I misunderstand something, I hope you could provide me with detailed guidance. Thank you very much.

https://github.com/apache/shardingsphere/blob/5.3.1/sql-parser/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLStatementSQLVisitor.java#L512

Reply:
Could you please help me confirm if my understanding is correct that the keywords passed into the`createPatternMatchingOperationSegment` function and the generated `BinaryOperationExpression` are only used for logical judgment and will not affect the rewritten SQL statements? 
Thank you in advance for your kind assistance.

```txt
patternMatchingOperator
    : LIKE
    | TILDE_TILDE_
    | NOT LIKE
    | NOT_TILDE_TILDE_
    | ILIKE
    | ILIKE_
    | NOT ILIKE
    | NOT_ILIKE_
    | SIMILAR TO
    | NOT SIMILAR TO
    | TILDE_
    | NOT_ TILDE_
    | TILDE_ ASTERISK_
    | NOT_ TILDE_ ASTERISK_
    ;
```
https://github.com/apache/shardingsphere/blob/5.3.1/sql-parser/dialect/postgresql/src/main/antlr4/imports/postgresql/BaseRule.g4#L530

Reply:
@wangtiga There seems to be some problems with the processing of LIKE in MySQL. Usually, the SQL parsing engine should not change the content of SQL, but standardize the SQL AST during the SQL Bind stage. It's going to take a lot of extra work to get this done though, so let's merge this PR first to make sure the functionality is correct.

Reply:
Besides, can you take a look at openGauss database? It maybe have the same problem.

Reply:
ok

Reply:
> There seems to be some problems with the processing of LIKE in MySQL. Usually, the SQL parsing engine should not change the content of SQL, but standardize the SQL AST during the SQL Bind stage. It's going to take a lot of extra work to get this done though, so let's merge this PR first to make sure the functionality is correct.

Thank you for your patient explanation. The similar issue in `OpenGaussStatementSQLVisitor` has been fixed and committed. The problem of directly assigning `LIKE` or `NOT LIKE` to `BinaryOperationExpression.operator` exists in MySQL and SQLServer as well.

https://github.com/apache/shardingsphere/blob/5.3.1/sql-parser/dialect/sqlserver/src/main/java/org/apache/shardingsphere/sql/parser/sqlserver/visitor/statement/impl/SQLServerStatementSQLVisitor.java#L435